### ISSUE TITLE
read_i2c_block_data() needs updating for CircuitPython 6 - "extra keyword arguments given" error

### ISSUE THREAD

**Issue by kevinjwalters** [pimoroni/circuitpython_adapter#1]
The combined plotted program spits this on CircuitPython 6.0.0 on a Feather nRF52840 Express with Enviro+ FeatherWing attached but no PMS5003:

```
PMS5003 Read Timeout: Failed to read start of frame byte
You probably don't have a pms5003 connected, continuing without particulate logging
Traceback (most recent call last):
  File "code.py", line 60, in <module>
  File "lib/pimoroni_ltr559/__init__.py", line 218, in __init__
  File "lib/i2cdevice/__init__.py", line 230, in get
  File "lib/i2cdevice/__init__.py", line 183, in read_register
  File "lib/i2cdevice/__init__.py", line 288, in _i2c_read
  File "lib/pimoroni_circuitpython_adapter/__init__.py", line 46, in read_i2c_block_data
  File "lib/pimoroni_circuitpython_adapter/__init__.py", line 40, in read_i2c_block_data
TypeError: extra keyword arguments given
```

This is due to the removal of the stop kwarg from `I2C.writeto()` in [CircuitPython 6.0.0](https://forums.adafruit.com/viewtopic.php?f=60&t=171913) (see https://github.com/adafruit/circuitpython/issues/2082 for details).

The [documentation for this on 6.0.x](https://circuitpython.readthedocs.io/en/6.0.x/shared-bindings/busio/index.html?highlight=i2c%20write#busio.I2C) is currently _incorrect_. I confirmed that with @dhalbert.

**Comment by kevinjwalters** [comment]
The recommened (more efficient) replacement `writeto_then_readfrom` is present in 5.3.x and 6 but _not_ 4.x:

- https://circuitpython.readthedocs.io/en/4.x/shared-bindings/busio/I2C.html?highlight=writeto_then_readfrom(#busio.I2C 
- https://circuitpython.readthedocs.io/en/5.3.x/shared-bindings/busio/I2C.html?highlight=writeto_then_readfrom(#busio.I2C.writeto_then_readfrom
- https://circuitpython.readthedocs.io/en/6.0.x/shared-bindings/busio/index.html?highlight=writeto_then_readfrom#busio.I2C.writeto_then_readfrom

**Comment by dglaude** [comment]
If you want to avoid this problem on your Enviro+, you may want to use this library for your LTR559:
https://github.com/pimoroni/Pimoroni_CircuitPython_LTR559/

There is a bit of discussion on the evolution of Pimoroni libraries to be a better CP citizen:
* https://github.com/pimoroni/EnviroPlus-FeatherWing/issues/17
* https://github.com/pimoroni/EnviroPlus-FeatherWing/issues/5

(It does not fix the problem with this library, but it might be the way to go)

**Comment by kevinjwalters** [comment]
Thanks, I'll have a look. I just put PR #2 in which I think fixes all of this.

**Comment by kevinjwalters** [comment]
Fixed with #2
